Fix `go vet` issues

**Bug description**
This is to address below issues reported by `go vet `
- cmd/cli/plugin/login/main.go:367:4: unreachable code
- pkg/v1/tkg/test/tkgctl/shared/tkrresolver_cc.go:335:26: Sprintf format %s has arg mdIndex of wrong type int

**Affected product area (please put an X in all that apply)**

- ( ) APIs
- ( ) Addons
- (x) CLI
- ( ) Docs
- ( ) IAM
- ( ) Installation
- ( ) Plugin
- ( ) Security
- ( ) Test and Release
- ( ) User Experience
- ( ) Developer Experience

**Expected behavior**

**Steps to reproduce the bug**

**Version** (include the SHA if the version is not obvious)

**Environment where the bug was observed (cloud, OS, etc)**

**Relevant Debug Output (Logs, manifests, etc)**

--- a/cmd/cli/plugin/login/main.go
+++ b/cmd/cli/plugin/login/main.go
@@ -361,10 +361,6 @@ func createServerWithEndpoint() (server *configv1alpha1.Server, err error) {
 		if err != nil {
 			log.Fatalf("Error creating kubeconfig with tanzu pinniped-auth login plugin: %v", err)
 			return nil, err
-			// IsSupervisorClusterIgnorePort only returns an error if the
-			// underlying HTTP GET returned one. Non-200 responses will return
-			// nil errors.
-			log.Warning("Unable to determine if endpoint was a vSphere Supervisor")
 		}
 		if isVSphereSupervisor {
 			log.Info("Detected a vSphere Supervisor being used")

--- a/pkg/v1/tkg/test/tkgctl/shared/tkrresolver_cc.go
+++ b/pkg/v1/tkg/test/tkgctl/shared/tkrresolver_cc.go
@@ -332,7 +332,7 @@ func validateControlPlanceTKRDataFields(cluster *clusterv1.Cluster, result data.
 func validateMDTKRDataFields(cluster *clusterv1.Cluster, mdIndex int, result data.Result) {
 	var tkrData clusterdata.TKRData
 	err := topology.GetMDVariable(cluster, mdIndex, "TKR_DATA", &tkrData)
-	Expect(err).To(BeNil(), fmt.Sprintf("failed to get the TKR_DATA variable for MD[%s]", mdIndex))
+	Expect(err).To(BeNil(), fmt.Sprintf("failed to get the TKR_DATA variable for MD[%v]", mdIndex))
 
 	osImageResolved := tkrData[cluster.Spec.Topology.Version].Labels[runv1.LabelOSImage]
 	Expect(osImageResolved).ToNot(BeEmpty(), fmt.Sprintf("MD[%d]'s TKR_DATA[%s] is missing label:'%s' ", mdIndex, cluster.Spec.Topology.Version, runv1.LabelOSImage))

